src/server/debug/server/server.go

```diff
@@ -101,9 +102,53 @@ func NewDebugServer(env Env) debug.DebugServer {
 		logLevel:      log.LogLevel,
 		grpcLevel:     log.GRPCLevel,
 	}
+	ctx := pctx.Background("debug server")
+	go s.logDBState(ctx)
 	return s
 }
 
+func (s *debugServer) logDBState(ctx context.Context) {
+	ticker := time.NewTicker(3 * time.Second)
+	queries := map[string]string{
+		"idle queries": `
+                  SELECT current_timestamp - query_start AS runtime,
+                         datname,
+                         usename,
+                         client_addr,
+                         backend_xid,
+                         query
+		  FROM pg_stat_activity WHERE state = 'idle' AND current_timestamp - query_start > interval '10 seconds' ORDER BY runtime DESC;`,
+		"active queries": `
+                  SELECT current_timestamp - query_start AS runtime,
+                         datname,
+                         usename,
+                         client_addr,
+                         backend_xid,
+                         query
+		  FROM pg_stat_activity WHERE state != 'idle' AND current_timestamp - query_start > interval '10 seconds' ORDER BY runtime DESC;`,
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			log.Error(pctx.TODO(), "log db context cancelled", zap.Error(context.Cause(ctx)))
+			return
+		case <-ticker.C:
+			for msg, query := range queries {
+				rows, err := s.database.QueryContext(ctx, query)
+				if err != nil {
+					log.Error(ctx, fmt.Sprintf("query %s", msg), zap.Error(err))
+				}
+				var buf bytes.Buffer
+				if err := s.writeRowsToJSON(rows, &buf); err != nil {
+					log.Error(ctx, fmt.Sprintf("write results for  %s", msg), zap.Error(err))
+				}
+				log.Info(ctx, msg, zap.String("txns", buf.String()))
```

Comment:
This is going to be JSON inside of JSON.  Not the end of the world but a little ugly.  Ideally this would return [][]string or something like that, and then logged with zap.Any().  Then you can get at the records without parsing JSON twice